packages/twenty-front/src/modules/activities/emails/components/EmailThreadMessage.tsx

```diff
@@ -22,21 +22,52 @@ const StyledThreadMessageHeader = styled.div`
 `;
 
 type EmailThreadMessageProps = {
-  id: string;
   body: string;
   sentAt: string;
-  from: MockedEmailUser;
-  to: MockedEmailUser[];
+  participants: EmailThreadMessageParticipant[];
+};
+
+const getDisplayNameFromParticipant = (
+  participant: EmailThreadMessageParticipant,
+) => {
+  if (participant.displayName) {
+    return participant.displayName;
+  }
+
+  if (participant.person) {
+    return `${participant.person?.name?.firstName} ${participant.person?.name?.lastName}`;
+  }
+
+  if (participant.workspaceMember) {
+    return `${participant.workspaceMember?.name?.firstName} ${participant.workspaceMember?.name?.lastName}`;
+  }
+
+  if (participant.handle) {
+    return participant.handle;
+  }
+
+  return 'Unknown';
 };
 
 export const EmailThreadMessage = ({
   body,
   sentAt,
-  from,
+  participants,
 }: EmailThreadMessageProps) => {
-  const { displayName, avatarUrl } = from;
   const [isOpen, setIsOpen] = useState(false);
 
+  const from = participants.find((participant) => participant.role === 'from');
+  const to = participants.filter((participant) => participant.role === 'to');
+
+  if (!from || to.length === 0) {
+    return null;
+  }
+
+  const displayName = getDisplayNameFromParticipant(from);
+
+  const avatarUrl =
```

Comment:
Question: Do we handle auto-generated avatar using the displayName if avatarUrl is absent in EmailThreadMessageSender?

Reply:
Yes! An empty string (which is the default value) will generate an avatar with the first letter of the display name